minetest/minetest
# Bloom

This PR adds exposure control and bloom effect to the game.

The following settings are introduced:
* `exposure_factor` emulates the amount of light that is captured on screen. Default: 1.0, min: 0.1, max: 10.0
* `enable_bloom` turns bloom effect on and off
* `bloom_intensity` controls the amount of bloom on screen. Default: 0.05, min: 0.01 max: 1.0
* `bloom_radius` controls the size of the blur convolution kernel. Default: 16, min: 1, max: 64

## To do

This PR is Ready for Review.

## How to test

1. Turn bloom on
2. Start any game
3. Find scene where source of light overlaps a dark object. Examples:
  * Looking at the sky/sun from a dark cave
  * Looking at lava from around a corner
  * Lava in front of obsidian
  * Forest skyline
4. Notice that the light bleeds from the light source to the dark area

Experiments:
* Type `/set exposure_factor 2` - entire scene becomes lighter
* Type `/set bloom_radius 32` - light bleeds further into the dark areas
* Type `/set bloom_intensity 0.15` - the intensity of bloom effect is higher.
